bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/parallacticangle/ParallacticAngleControls.scala

```diff
@@ -237,54 +243,17 @@ class ParallacticAngleControls(isPaUi: Boolean) extends GridBagPanel with Publis
     for {
       e      <- editor
       ispObs <- Option(e.getContextObservation)
-    } {
-      val spObs = ispObs.getDataObject.asInstanceOf[SPObservation]
-      val sameNight = spObs.getSchedulingBlock.asScalaOpt.exists(_.sameObservingNightAs(sb))
-
-      // This is the action that will update the scheduling block
-      val updateSchedBlock: IO[Unit] =
-        for {
-          _ <- IO(spObs.setSchedulingBlock(ImOption.apply(sb)))
-          _ <- IO(ispObs.setDataObject(spObs))
-        } yield ()
-
-      // Ok, this is an IO action that goes and fetches the ephemerides and returns ANOTHER action
-      // that will actually update the model and clear out the glass pane UI.
-      val fetch: IO[IO[Unit]] =
-        if (sameNight) IO(ispObs.silentAndLocked(updateSchedBlock))
-        else EphemerisUpdater.refreshEphemerides(ispObs, sb.start, e.getWindow).map { completion =>
-
-          // This is the action that will update the model
-          val up: IO[Unit] =
-            for {
-              _ <- updateSchedBlock
-              _ <- completion
-            } yield ()
-
-          // Here we bracket the update to disable/enable events and grab/release the program lock
-          ispObs.silentAndLocked(up)
-
+    } SchedulingBlockUpdate.runWithCallback(
```

Comment:
I took the code that followed the `for` in `ParallacticAngleControls` and moved it to a `SchedulingBlockUpdate` so it could be reused by the bulk update menu item.  It also needed to work with multiple observations of course.